ReverseTicTacToe/UI/ConsoleUI.cs

```diff
@@ -33,17 +29,145 @@ public  void Init()
         }
         public  void Run()
         {
+            
             WriteLine("Welcome to Reverse Tic Tac Toe!" + Environment.NewLine);
-            getBoardSizeFromUser();
-            getGameTypeFromUser();
-            m_Game = new Game(m_BoardSize, m_GameType);
-            while(true)
+            int BoardSize =  getBoardSizeFromUser();
+            Game.eGameTypes GameType = getGameTypeFromUser();
+            m_Game = new Game(BoardSize, GameType);
+           // while(true)
+            //{
+            printGameBoard();
+            playTurn();
+            ReadLine();
+            
+            //}
+        }
+
+        private void playTurn()
+        {
+            int row = 0;
+            int col = 0;
+            //int return status: continue, place_taken, finished game
+            string turnMsg = GetTurnMessage();
+            
+            if(m_Game.CurrentTurn == 1 || m_Game.GameType == Game.eGameTypes.PVP)
+            {
+                string userTurnInput;
+                do
+                {
+                    WriteLine(turnMsg);
+                    userTurnInput = ReadLine();
+                }
+                while(!turnInputIsValid(userTurnInput, out row, out col));
+            }
+            if (row == -1 && col == -1)
+            {
+                //quit game
+            }
+            m_Game.MakeTurn(row, col);
+        }
+
+        private bool turnInputIsValid(string i_UserTurnInput, out int i_Row, out int i_Col)
+        {
+            i_Row = -1;
+            i_Col = -1;
+            const bool v_IsValid = true;
+            bool turnInputIsValid = !v_IsValid;
+            char[] charsToTrim = { ' ', '\t' };
+            i_UserTurnInput = i_UserTurnInput.Trim(charsToTrim);
+            if(i_UserTurnInput == "Q")
+            {
+                turnInputIsValid = v_IsValid;
+            }
+            else if (i_UserTurnInput.Length == 3 && i_UserTurnInput[1] == ',')
+            {
+                char[] charArray = i_UserTurnInput.ToCharArray();
+                if(char.IsDigit(charArray[0]) && char.IsDigit(charArray[2]) && charArray[1] == ',')
```

Comment:
I don't fully understand what's happening in within that "if ", can you maybe add some documantation here?

Reply:
doc added